include/pybind11/detail/smart_holder_type_casters.h

```diff
@@ -106,6 +106,34 @@ class modified_type_caster_generic_load_impl {
         return false;
     }
 
+    bool try_as_void_ptr_capsule(handle src) {
+        std::string type_name = cpptype->name();
+        detail::clean_type_id(type_name);
+
+        // Convert `a::b::c` to `a_b_c`
+        size_t pos = type_name.find("::");
+        while (pos != std::string::npos) {
+            type_name.replace(pos, 2, "_");
+            pos = type_name.find("::", pos);
+        }
+
+        std::string as_void_ptr_function_name = "as_" + type_name;
+        if (hasattr(src, as_void_ptr_function_name.c_str())) {
+          auto void_ptr_capsule = reinterpret_borrow<object>(
```

Comment:
We have C++ wrappers for methods and Capsules. Any reason we aren't using them here? Speed presumably? If so leave comment. If not, can we host any of the functions to that Capsule class if they make sense there?

Reply:
Thanks for your suggestions! I am new to pybind11 and did not realize that I can use these wrappers.